### PR TITLE
2023-08-07: Add weekly report

### PR DESCRIPTION
**Pull request by P-E-P**
I'm not used to data collection from issues/milestone. I'll investigate further to collect issues/task completion during the past week. In the meantime, most of it is here and can be reviewed.

### CODE REVIEW

File: 2023-08-07-report.org [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+Thanks again to [[https://opensrcsec.com/][Open Source Security, inc]] and [[https://www.embecosm.com/][Embecosm]] for their ongoing support for this project.
+
+*** Milestone Progress
+
+1. Name resolution rework
+  1. Pushed many improvements over the existing code
+  2. Early
+    1. Most of early name resolution has been merged
+    3. Procedural macro name resolution are almost finished
```
Comment (dkm) [inline 2023-08-07-report.org:15]: Missing the 2. here, or most probably, 3 should be 2?

Comment (P-E-P) [reply 2023-08-07-report.org:15]: Yeah good catch. Do you see anything we should add to this section btw ? Maybe a line for your commits ?

Comment (dkm) [reply 2023-08-07-report.org]: My changes are not really part of any milestone, so not really sure.
Maybe we can add "Minor refactor of the macro transcriber functions to use the newly added tl::optional<>".
But I'll leave it to you to decide if that's worth a mention.

**Comment by P-E-P** [comment]
@CohenArthur this PR missed accurate data for the whole milestone/GitHub data thing.

**Comment by CohenArthur** [comment]
@P-E-P that's okay, no worries. It's only a weekly report and I wanted a base for today's report :) it will be all good in the monthly report. Now go back to vacation!

**Pull request merged**